### PR TITLE
[flang][OpenMP] Convert unique clauses in ClauseProcessor

### PR DESCRIPTION
**Pull request by kparzysz**
Temporarily rename old clause list to `clauses2`, old clause iterator to `ClauseIterator2`.
Change `findUniqueClause` to iterate over `omp::Clause` objects, modify all handlers to operate on 'omp::clause::xyz` equivalents.

[Clause representation 2/6]

### CODE REVIEW

File: flang/lib/Lower/OpenMP.cpp [hunk 1]
```diff
@@ -2639,54 +2641,50 @@ bool ClauseProcessor::processFinal(Fortran::lower::StatementContext &stmtCtx,
...
 
 bool ClauseProcessor::processOrdered(mlir::IntegerAttr &result) const {
-  if (auto *orderedClause = findUniqueClause<ClauseTy::Ordered>()) {
+  if (auto *clause = findUniqueClause<omp::clause::Ordered>()) {
     fir::FirOpBuilder &firOpBuilder = converter.getFirOpBuilder();
     int64_t orderedClauseValue = 0l;
-    if (orderedClause->v.has_value()) {
-      const auto *expr = Fortran::semantics::GetExpr(orderedClause->v);
-      orderedClauseValue = *Fortran::evaluate::ToInt64(*expr);
+    if (clause->v.has_value()) {
```
Comment (skatrak) [inline flang/lib/Lower/OpenMP.cpp:2686]: Nit: Braces not necessary.

**Review by skatrak — COMMENTED** [review]
Thanks, LGTM.

File: flang/lib/Lower/OpenMP.cpp (hunk 1 above)
Comment (kparzysz) [reply flang/lib/Lower/OpenMP.cpp:2686]: Done

**Review by kiranchandramohan — APPROVED** [review]
Mostly mechanical changes. LGTM. Please wait for @skatrak

**Review by skatrak — APPROVED** [review]
LGTM.

**Review by tblah — APPROVED** [review]
LGTM

**Pull request merged**